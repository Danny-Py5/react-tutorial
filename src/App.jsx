// react useEffect Hook ---> 4:21:00
import "./globals.css";
import { useState, useEffect } from "react";

export default function App() {
  const [count, setCount] = useState(0);
  const [sync, setSync] = useState(false);

  useEffect(() => {
    document.title = `UseEffect Hook - ${count} times`;
  }, [sync]); // this will run when the count changes

  return (
    <div>
      <h1>React useEffect Hook</h1>
      <p>You clicked {count} times </p>
      <button onClick={() => setCount((prevCount) => prevCount + 1)}>
        Increment
      </button>
      <button onClick={() => setSync((preSync) => !preSync)}>Sync</button>
    </div>
  );
}

// ---------------------------------------------------

// state and Array --->  3:30:00
// import "./globals.css";
// import UserProfile from "./state/UserProfile.jsx";
// import { PracticeMutationAndImmutability } from "./state/practice-mutation-immutability/PracticeMutationAndImmutability.jsx";

// import { CreateUser } from "./state/CreateUser.jsx";

// import { useState } from "react";

// export default function App() {
//   const [users, setUsers] = useState([
//     {
//       id: 1,
//       username: "daniel",
//       email: "olatundedaniel@gmail.com",
//     },
//     {
//       id: 2,
//       username: "kemisola",
//       email: "kemisolasajo@gmail.com",
//     },
//     {
//       id: 3,
//       username: "Samuel",
//       email: "samuel@gmail.com",
//     },
//     {
//       id: 4,
//       username: "grace",
//       email: "grace@gmail.com",
//     },
//   ]);
//   // console.log(users);
//   return (
//     <div>
//       <h2>User Profile</h2>
//       <CreateUser setUsers={setUsers} />
//       <br /> <br />
//       {
//         <div className="User-details">
//           {users.map((user) => {
//             return (
//               <UserProfile key={user.id} user={user} setUsers={setUsers} />
//             );
//           })}
//         </div>
//       }
//       {/* <PracticeMutationAndImmutability /> */}
//     </div>
//   );
// }
// continue from here // 3:41:00

// ------------------------------------------------
// // React state: 2:54:40;
// import CounterApp from "./state/counter-app/CounterApp.jsx";
// import "./globals.css";
// import { RegisterUser } from "./state/RegisterUser.jsx";
// import { RegisterUserState2 } from "./state/RegisterUserState2.jsx";

// export default function App() {
//   return (
//     // <>
//     //   <RegisterUserState2 />
//     // </>
//     <CounterApp />
//   );
// }

// // event handeling 2:25:18 --->
// import "./globals.css";
// import LoginForm from "./event-handeling/LoginForm.jsx";

// export default function App() {
//   return (
//     <div>
//       <LoginForm />
//     </div>
//   );
// }

// ------------------------------------------------
// // #conditional rendering  <= 2:25
// import "./globals.css";
// import MorkAuthentication from "./conditional-rendering/MorkAuthentication.jsx";
// import AccountStatus from "./conditional-rendering/AccountStatus.jsx";

// export default function App() {
//   return (
//     <>
//       <MorkAuthentication />
//       <AccountStatus />
//     </>
//   );
// }

// ------------------------------------------------
// import "./globals.css";
// import { User } from "./renderiing-list/User.jsx";

// export default function App() {
//   return (
//     <div>
//       <User />
//     </div>
//   );
// }
// ------------------------------------------------
// import "./globals.css";
// import { RenderArticleComponenet } from "./react-css/RenderArticleComponenet.jsx";

// export default function App() {
//   return (
//     <div>
//       <RenderArticleComponenet />
//     </div>
//   );
// }
// ------------------------------------------------
// import { User } from "./fragment/User.jsx";
// import MyHobby from "./fragment/MyHobby.jsx";

// // using fragment to render all component to the root
// export default function App() {
//   return (
//     // grouping components to be direct child of the root
//     <>
//       <User name="daniel" age={20} />
//       <br />
//       <br />
//       <MyHobby />
//     </>
//   );
// }
// ------------------------------------------------
// export default function App() {
//   return (
//     <div>
//       <UserProfile
//         username="Daniel Olatunde"
//         age={20}
//         favFoods={[
//           {
//             name: "rice",
//             price: "400",
//           },
//           {
//             name: "beans",
//             price: 500,
//           },
//           {
//             name: "garri",
//             price: 200,
//           },
//         ]}
//       />
//     </div>
//   );
// }
// ------------------------------------------------
// import { UserProfileCard } from "./practice-component-and-props/UserProfileCard.jsx";
// export default function App() {
//   return (
//     <div>
//       <h1>component and props practice</h1>

//       <UserProfileCard
//         name="Daniel Olatunde"   // this is a prop
//         avatarUrl="http://dannypyAvataUrl.com" // this is a prop
//         email="danielolatunde@gmail.com"  // this is a prop
//         phone="0816552344" // this is a prop
//         location="Rubber Estate, Ondo State Nigeria" // this is a prop
//         bio="curious about coding" // this is a prop
//         isOnline={true} // this is a prop
//         height="50cm" // this is a prop
//         favouriteFoods={["Rice", "Beans", "Egg", "milk"]} // this is a prop
//       />
//     </div>
//   );
// }
// ------------------------------------------------
// import { Product } from "./component/Product.jsx";
// export default function App() {
//   return (
//     <div>
//       <h1>This is the root component</h1>

//       <br />
//       <Product />
//     </div>
//   );
// }
