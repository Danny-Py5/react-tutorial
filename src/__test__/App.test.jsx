import { describe, expect, it } from "vitest";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import App from "../App.jsx";

describe("App", () => {
  it("should render user", () => {
    const result = render(<App />);
    expect(result.container).toMatchSnapshot();
  });

  // to save long (it) test description short and simple, grouped into another test suit. We know that all the tests in the suit are for edit button
  describe("Edit button is clicked", () => {
    it("Should display the update button", async () => {
      render(<App />);
      const editButton = screen.getByRole("button", { name: "Edit" });
      await userEvent.click(editButton);
      const saveButton = screen.getByRole("button", { name: "Update" });
      expect(saveButton).toBeInTheDocument();
    });

    // test description would have being this (should display the userName and the name input field when the edit button is clicked)
    it("should display the userName input field", async () => {
      render(<App />);
      const editButton = screen.getByRole("button", { name: "Edit" });
      await userEvent.click(editButton);
      const userNameInput = screen.getByRole("textbox", { name: "username" });
      expect(userNameInput).toBeInTheDocument();
    });
  });
});

// my questions
/* 
tell me differences btw thing i import from vitest and from testing-react/react 

what vitest imports are for and do and also what imports from testing-library/react are for and what they do?


*/
