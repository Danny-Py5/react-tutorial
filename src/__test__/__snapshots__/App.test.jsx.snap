// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render user 1`] = `
<div>
  <div>
    <button>
      Edit
    </button>
    <button>
      Delete
    </button>
    <p>
      Id: 
      <span>
        1
      </span>
    </p>
    <p>
      Name: 
      <span>
        Daniel Olatunde
      </span>
    </p>
    <p>
      UserName:
       
      <span>
        Dannypy
      </span>
    </p>
  </div>
</div>
`;
